src/components/SerialConsole/SerialConsole.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import XTerm from './XTerm';
+
+const defaultMessages = {
+  textDisconnect: 'Disconnect',
+  textDisconnected:
+    'Disconnected from serial console. Click the Reconnect button.',
+  textLoading: 'Loading ...',
+  textReconnect: 'Reconnect'
+};
+
+class SerialConsole extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusTerminal = this.focusTerminal.bind(this);
+    this.onResetClick = this.onResetClick.bind(this);
+    this.onDisconnectClick = this.onDisconnectClick.bind(this);
+
+    this.msg = this.msg.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.onConnect();
+  }
+
+  componentWillUnmount() {
+    this.props.onDisconnect();
+  }
+
+  msg(key) {
```

Comment:
I don't really get it... why not just use `defaultProps and specify the value`? Maybe someone can explain this...

Reply:
Based o a previous review comment, I expect this is a pattern in this project. If not, the former direct use of `props` seems to me more straightforward.

Reply:
done, defaultProps used instead